service-clients-gen/src/main/java/com/flipkart/poseidon/serviceclients/generator/ServiceGenerator.java

```diff
@@ -133,6 +134,12 @@ private void addInterfaceFields(ServiceIDL serviceIdl, JCodeModel jCodeModel, JD
         invocation.arg(JExpr.lit(serviceIdl.getVersion().getMinor()));
         invocation.arg(JExpr.lit(serviceIdl.getVersion().getPatch()));
         versionField.init(invocation);
+
+        if (serviceIdl.getService().getObjectMapperClass() != null) {
+            fieldModifier = JMod.PUBLIC | JMod.STATIC | JMod.FINAL;
```

Comment:
I have added this in CartServiceClient(interface) which cann't be protected. I will remove static here.